src/components/BadgeOverlay/index.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { ReactNode } from 'react';
+import PropTypes from 'prop-types';
+
+import abbreviateNumber from '../AvatarGroup/helpers/AbbreviateNumber';
+import { BaseProps } from '../types';
+import { positionOverlay, variantOverlay, overlapOverlay } from './types';
+import {
+    StyledOverlayContainer,
+    StyledBadge,
+    StyledBadgeOverlayText,
+} from './styled';
+import RenderIf from '../RenderIf';
+
+interface Props extends BaseProps {
+    children?: ReactNode;
+    isHidden?: boolean;
+    overlap?: overlapOverlay;
+    position?: positionOverlay;
+    value?: string | number;
+    variant?: variantOverlay;
+}
+
+/**
+ * Badge Overlay is used to show small numerical value or status descriptors
+ * for UI elements. Normally appears in proximity to notifications or
+ * user avatars with eye-catching appeal, typically displaying unread messages count.
+ */
+
+const BadgeOverlay: React.FC<Props> = ({
+    children,
+    isHidden = false,
+    position = 'top-right',
+    value = '',
+    variant = 'error',
+    overlap = 'rectangle',
+}) => {
+    const total =
+        typeof value === 'string'
+            ? isNaN(parseInt(value, 10))
+                ? value
+                : abbreviateNumber(parseInt(value, 10))
+            : value;
```

Comment:
if we need to do some check it should be part of the helper, but instead, use the same as in the react rainbow components since it works properly